ca/ca.go

```diff
@@ -186,18 +186,52 @@ func (ca *CAImpl) newIntermediateIssuer(root *issuer, intermediateKey crypto.Sig
 	}, nil
 }
 
-func (ca *CAImpl) newChain(intermediateKey crypto.Signer, intermediateSubject pkix.Name, subjectKeyID []byte) *chain {
+// newChain generates a new issuance chain, including a root certificate and numIntermediates intermediates (at least 1).
+// The first intermediate will use intermediateKey, intermediateSubject and subjectKeyId.
+// Any intermediates between the first intermediate and the root will have their keys and subjects generated automatically.
+func (ca *CAImpl) newChain(intermediateKey crypto.Signer, intermediateSubject pkix.Name, subjectKeyID []byte, numIntermediates int) *chain {
+	if numIntermediates <= 0 {
+		panic("At least one intermediate must be present in the certificate chain")
+	}
 	root, err := ca.newRootIssuer()
 	if err != nil {
 		panic(fmt.Sprintf("Error creating new root issuer: %s", err.Error()))
 	}
-	intermediate, err := ca.newIntermediateIssuer(root, intermediateKey, intermediateSubject, subjectKeyID)
+
+	// The last N-1 intermediates build a path from the root to the leaf signing certificate.
+	// If numIntermediates is only 1, then no intermediates will be generated here.
+	prev := root
+	intermediates := make([]*issuer, numIntermediates)
+	for i := 0; i < numIntermediates-1; i++ {
+		k, ski, err := makeKey()
+		if err != nil {
+			panic(fmt.Sprintf("Error creating new intermediate issuer: %v", err))
+		}
+		intermediate, err := ca.makeRootCert(k, pkix.Name{
```

Comment:
Why aren't you using `newIntermediateIssuer` here?
```suggestion
		intermediates[i], err := ca.newIntermediateIssuer(prev, k, pkix.Name{
```